src/modules/IsnsServer.rb

```diff
@@ -351,7 +351,9 @@ def isnsadm(params, ret_result = false)
 
     def isnsadm_control
       if !@isctrlnode
-        @isctrlnode = isnsadm("--register control")
+        res = isnsadm("--register control", ret_result = true)
+        @isctrlnode = res["exit"] == 0
```

Comment:
I do not see from commit message why this should fix it as you basically do what isnsadm do inside itself. see https://github.com/yast/yast-isns/pull/9/files#diff-1b6bb3b963d2429af986b5c9bbbebaa9R348 it do same comparison, so why it does not work?